testing/camunda-process-test-example/src/test/java/io/camunda/InvoiceApprovalNestedTest.java

```diff
@@ -0,0 +1,262 @@
+/*
+ * Copyright © 2017 camunda services GmbH (info@camunda.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.camunda;
+
+import static io.camunda.process.test.api.CamundaAssert.assertThatProcessInstance;
+import static io.camunda.process.test.api.CamundaAssert.assertThatUserTask;
+import static io.camunda.process.test.api.assertions.ElementSelectors.byId;
+import static io.camunda.process.test.api.assertions.UserTaskSelectors.byElementId;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.verify;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.camunda.client.CamundaClient;
+import io.camunda.process.test.api.CamundaProcessTestContext;
+import io.camunda.process.test.api.CamundaSpringProcessTest;
+import io.camunda.process.test.api.mock.JobWorkerMockBuilder.JobWorkerMock;
+import io.camunda.services.ArchiveService;
+import io.camunda.services.WiredLegacyException;
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.Map;
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.bean.override.mockito.MockitoBean;
+
+@SpringBootTest(
+    properties = {
+      "camunda.client.worker.defaults.enabled=false", // disable job workers
+      // but enable the normal glue code that just delegates to the ArchiveService
+      "camunda.client.worker.override.archive-invoice.enabled=true",
+    })
+@CamundaSpringProcessTest
+@DisplayName("Invoice Approval Process Test")
+public class InvoiceApprovalNestedTest {
```

Comment:
🔧 I see no need to add this test class in addition to `ProcessNestedUnitTest`. One example is enough to demonstrate the usage of nested classes.

Reply:
see above